### PR TITLE
Consensus fix + throw exception on accepting cb with missing height

### PR DESCRIPTION
**Pull request by wkoszycki**

### CODE REVIEW

File: src/main/scala/org/constellation/primitives/EdgeDAO.scala [hunk 1]
```diff
@@ -378,8 +380,7 @@ class ThreadSafeSnapshotService(concurrentTipService: ConcurrentTipService) {
           }
           _ <- cb.storeSOE()
           maybeHeight <- calculateHeight(checkpoint)
-          metricKey = if (maybeHeight.isEmpty) Metrics.heightEmpty else Metrics.heightNonEmpty
-          _ <- dao.metrics.incrementMetricAsync(metricKey)
+          _ <- if(maybeHeight.isEmpty) dao.metrics.incrementMetricAsync(Metrics.heightEmpty).flatMap(_ => IO.raiseError(MissingHeightException(cb))) else IO.unit
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/primitives/EdgeDAO.scala]: IMO raiseError is the main effect you want to trigger, thus:
`IO.raiseError(MissingHeightException(cb).flatTap(_ => dao.metrics.incrementMetricAsync(Metrics.heightEmpty)`

File: src/main/scala/org/constellation/primitives/EdgeDAO.scala [hunk 2]
```diff
@@ -394,8 +395,8 @@ class ThreadSafeSnapshotService(concurrentTipService: ConcurrentTipService) {
     }
     acceptCheckpoint.recoverWith {
       case err =>
-        IO.shift *> dao.metrics.incrementMetricAsync("acceptCheckpoint_failure")
-        IO.raiseError(err) //propagate to upper levels
+        dao.metrics.incrementMetricAsync("acceptCheckpoint_failure")
+            .flatTap(_ => IO.raiseError(err)) //propagate to upper levels
```
Comment (marcinwadon) [inline src/main/scala/org/constellation/primitives/EdgeDAO.scala]: as above

File: src/main/scala/org/constellation/primitives/EdgeDAO.scala (hunk 1 above)
Comment (wkoszycki) [reply src/main/scala/org/constellation/primitives/EdgeDAO.scala]: error will be swallowed and not propagated to upper levels

File: src/main/scala/org/constellation/primitives/EdgeDAO.scala (hunk 2 above)
Comment (wkoszycki) [reply src/main/scala/org/constellation/primitives/EdgeDAO.scala]: error will be swallowed and not propagated to upper levels

**Pull request merged**